### PR TITLE
Add caliban to benchmarks

### PR DESCRIPTION
**Pull request by kyri-petrou**
Hey guys, just wanted to add Caliban (from Scala) to the benchmarks.

Just wanted to highlight a couple of things:

1. The warmup that was previously being done didn't seem enough for JDK apps so I added a few more rounds. The JVM is still not fully warm after it but I think it's _good enough_.
2. The benchmarks in this PR are from a 2 CPU machine (in GH CodeSpace), which show TC ahead. Given that using such a low number of cores gives a fairly big disadvantage to JVM apps, I also run them with a 16-core machine, and these are RPS results:

image

(PS the netflix one was crashing in this run for some reason)

Just as feedback, I think it might be good to add a range of machine cores to the benchmarks just to highlight where different languages / libraries might have some advantage over others

EDIT: My apologies I rerun the benchmark at 8CPU after submitting the PR. You can look at the 2CPU results in the commit prior

### CODE REVIEW

File: graphql/caliban/src/main/scala/Main.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+final class RequestHandler(interpreter: GraphQLInterpreter[Service, CalibanError]) {
+  private val contentTypeJson: Headers = Headers(Header.ContentType(MediaType.application.json).untyped)
+
+  def apply(request: Request): URIO[Service, Response] = {
+    for {
+      arr  <- request.body.asArray.orDie
+      resp <- interpreter.executeRequest(readFromArray[GraphQLRequest](arr))
+    } yield Response(Status.Ok, contentTypeJson, Body.fromChunk(Chunk.fromArray(writeToArray(resp))))
```
Comment (tusharmath) [inline graphql/caliban/src/main/scala/Main.scala:37]:
```suggestion
    } yield Response(Status.Ok, contentTypeJson, Body.fromChunk(Chunk.fromArray(writeToArray(resp)))).withServerTime
```
Just to make sure we are spec compliant.

**Review by tusharmath — CHANGES_REQUESTED** [review]

**Comment by tusharmath** [comment]
Can you run the benchmark again and push the results.

**Comment by kyri-petrou** [comment]
Eghh I think that even after adding the extra warmups, the server is nowhere near warm enough at lower CPU :(

First _recorded_ run (i.e., after warmups) at 4CPU:

```
Running 10s test @ http://localhost:8084/graphql
  4 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   148.82ms  226.51ms   1.39s    91.87%
    Req/Sec   303.11    125.76   580.00     62.64%
  10799 requests in 10.05s, 55.43MB read
Requests/sec:   1074.99
Transfer/sec:      5.52MB
```

3rd one:
```
Running 10s test @ http://localhost:8084/graphql
  4 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    28.59ms    9.23ms 117.71ms   75.91%
    Req/Sec     0.88k   179.45     1.32k    69.75%
  35124 requests in 10.02s, 180.28MB read
Requests/sec:   3506.39
Transfer/sec:     18.00MB
```

**Comment by tusharmath** [comment]
@kyri-petrou How is it in comparison to others?

> Just as feedback, I think it might be good to add a range of machine cores to the benchmarks just to highlight where different languages / libraries might have some advantage over others

Can you open an issue for this? We can definitely run it on a series cpus.

**Comment by kyri-petrou** [comment]
> @kyri-petrou How is it in comparison to others?

The netflix server seems to have the same issue. The Apollo / Go / Tailcall have the same performance across all 3 _measured_ runs.

> Can you open an issue for this? We can definitely run it on a series cpus.

Yeah can do. The other thing which might be good is to perhaps consider using GraalVM as the JDK. Our benchmarks in the Caliban repo which measure the internal performance only have a ~50-100% higher throughput running on GraalVM. I still don't know why our execution benefits _this much_ from GraalVM though.

**Comment by tusharmath** [comment]
@kyri-petrou 
I think this PR still has the older results.  If that's the case, can you run again and update the PR please?

Our benchmarks are typically done on a 4core machine, which ideal for a production environment also. A layer like this should be scaled horizontally and not vertically to be cost-effective. So it's preferred to have 2-4 core machines in general.

We could use Graalvm. Feel free to make that change, I doubt the performance will change significantly.

**Comment by kyri-petrou** [comment]
Hmm I'm not sure why tailcall is showing up as `null` though :/ let me try and rerun them

**Comment by kyri-petrou** [comment]
OK seems to have worked now

**Review by tusharmath — APPROVED** [review]

**Pull request merged**